### PR TITLE
Prepare v1.7.0 release with Metrics API/SDK GA.

### PR DESCRIPTION
**Pull request by esigo**

### CODE REVIEW

**Comment by lalitb** [comment]
Just to add -  This PR is also the meant to be  GA release for Metrics API and SDK. If there are PRs or issues which should block this release, comment below and we can discuss further here.

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,42 @@ Increment the:
...
+* [METRICS SDK] Validate Instrument meta data (name, unit, description) [#1713](https://github.com/open-telemetry/opentelemetry-cpp/pull/1713)
+* [DOCS] Document libthrift 0.12.0 doesn't work with Jaeger exporter [#1714](https://github.com/open-telemetry/opentelemetry-cpp/pull/1714)
+* [Metrics SDK] Add Monotonic Property to Sum Aggregation, and
+unit tests for Up Down Counter [#1675](https://github.com/open-telemetry/opentelemetry-cpp/pull/1675)
+* [Metrics SDK] Move Metrics Exemplar processing behind feature flag [#1710](https://github.com/open-telemetry/opentelemetry-cpp/pull/1710)
+* [Metrics API/SDK] Change Meter API/SDK to return nostd::unique_ptr
+ for Sync Instruments [#1707](https://github.com/open-telemetry/opentelemetry-cpp/pull/1707)
+* [BUILD] Add e2e test to asan & tsan CI [#1670](https://github.com/open-telemetry/opentelemetry-cpp/pull/1670)
+* [BUILD] Add otlp-grpc example bazel [#1708](https://github.com/open-telemetry/opentelemetry-cpp/pull/1708)
+* Fix debug log of OTLP HTTP exporter and ES log exporter [#1703](https://github.com/open-telemetry/opentelemetry-cpp/pull/1701)
```
Comment (ThomsonTan) [inline CHANGELOG.md:29]:
```suggestion
* Fix debug log of OTLP HTTP exporter and ES log exporter [#1703](https://github.com/open-telemetry/opentelemetry-cpp/pull/1703)
```

Comment (esigo) [reply CHANGELOG.md]: thanks

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,42 @@ Increment the:
...
 
+## [1.7.0] 2022-10-28
+
+* [METRICS SDK] Validate Instrument meta data (name, unit, description) [#1713](https://github.com/open-telemetry/opentelemetry-cpp/pull/1713)
+* [DOCS] Document libthrift 0.12.0 doesn't work with Jaeger exporter [#1714](https://github.com/open-telemetry/opentelemetry-cpp/pull/1714)
+* [Metrics SDK] Add Monotonic Property to Sum Aggregation, and
+unit tests for Up Down Counter [#1675](https://github.com/open-telemetry/opentelemetry-cpp/pull/1675)
+* [Metrics SDK] Move Metrics Exemplar processing behind feature flag [#1710](https://github.com/open-telemetry/opentelemetry-cpp/pull/1710)
+* [Metrics API/SDK] Change Meter API/SDK to return nostd::unique_ptr
+ for Sync Instruments [#1707](https://github.com/open-telemetry/opentelemetry-cpp/pull/1707)
```
Comment (lalitb) [inline CHANGELOG.md:26]:
```suggestion
 for Sync Instruments [#1707](https://github.com/open-telemetry/opentelemetry-cpp/pull/1707) which includes breaking change in the Metrics api and sdk.
```

Comment (esigo) [reply CHANGELOG.md:26]: thanks

File: CHANGELOG.md [hunk 3]
```diff
@@ -15,6 +15,42 @@ Increment the:
...
+* [METRICS SDK] Add timeout support to MeterContext::ForceFlush [#1673](https://github.com/open-telemetry/opentelemetry-cpp/pull/1673)
+* [DOCS] - Minor updates to OStream Metrics exporter documentation [#1679](https://github.com/open-telemetry/opentelemetry-cpp/pull/1679)
+* [DOCS] Fix:#1575 API Documentation for Metrics SDK and API [#1678](https://github.com/open-telemetry/opentelemetry-cpp/pull/1678)
+* [BUILD] Fixed compiler warnings [#1677](https://github.com/open-telemetry/opentelemetry-cpp/pull/1677)
+* [METRICS SDK] Fix threading issue between Meter::RegisterSyncMetricStorage
+ and Meter::Collect [#1666](https://github.com/open-telemetry/opentelemetry-cpp/pull/1666)
+* [METRICS SDK] Fix data race on MeterContext::meters_ [#1668](https://github.com/open-telemetry/opentelemetry-cpp/pull/1668)
+* [METRICS SDK] Fix observable Gauge metrics generation [#1651](https://github.com/open-telemetry/opentelemetry-cpp/pull/1651)
+* [BUILD] Detect ARCH=sparc in CMake [#1660](https://github.com/open-telemetry/opentelemetry-cpp/pull/1660)
+* Add user agent for OTLP http/grpc client [#1657](https://github.com/open-telemetry/opentelemetry-cpp/pull/1657)
```
Comment (ThomsonTan) [inline CHANGELOG.md:43]: Missing tag here,  `[SDK]`?

File: CHANGELOG.md [hunk 4]
```diff
@@ -15,6 +15,42 @@ Increment the:
...
+* [BUILD] Fix clang and gcc warnings [#1658](https://github.com/open-telemetry/opentelemetry-cpp/pull/1658)
+* [Metrics SDK] Add Metrics ExemplarFilter and ExemplarReservoir [#1584](https://github.com/open-telemetry/opentelemetry-cpp/pull/1584)
+* [LOGS SDK] Rename OnReceive to OnEmit [#1652](https://github.com/open-telemetry/opentelemetry-cpp/pull/1652)
+* [METRICS SDK] Fix Observable Gauge does not reflect updated values,
+and send the old value always [#1641](https://github.com/open-telemetry/opentelemetry-cpp/pull/1641)
+* [Metrics SDK] Change boundary type to double for Explicit Bucket Histogram Aggregation,
+and change default bucket range [#1626](https://github.com/open-telemetry/opentelemetry-cpp/pull/1626)
+* [METRICS SDK] Fix occasional Segfault with LongCounter instrument [#1638](https://github.com/open-telemetry/opentelemetry-cpp/pull/1638)
+* [BUILD] Bump vcpk to 2022.08.15 [#1633](https://github.com/open-telemetry/opentelemetry-cpp/pull/1633)
+* [BUILD] Bump gRPC to v1.48.1 for CMake Linux CI [#1608](https://github.com/open-telemetry/opentelemetry-cpp/pull/1608)
```
Comment (ThomsonTan) [inline CHANGELOG.md:53]: Add a notes section on the end to include the breaking changes for metrics preview customer?

File: sdk/src/version/version.cc [hunk 5]
```diff
@@ -9,16 +9,16 @@ namespace sdk
...
-const char *COMMIT_HASH     = "38012fab34faa50d5020959b68eba77ac485b508";
-const char *SHORT_VERSION   = "1.6.1";
+const int COUNT_NEW_COMMITS = 35;
+const char *BRANCH          = "pre_release_1.7.0";
+const char *COMMIT_HASH     = "dd7e257b6de71eeaf9e3149530962301705b9a0d";
+const char *SHORT_VERSION   = "1.7.0";
 const char *FULL_VERSION =
-    "1.6.0-NONE-NONE-27-pre_release_1.6.1-38012fab34faa50d5020959b68eba77ac485b508";
+    "1.7.0-NONE-NONE-35-pre_release_1.7.0-dd7e257b6de71eeaf9e3149530962301705b9a0d";
 const char *BUILD_DATE = "Thu Sep 22 18:05:21 UTC 2022";
```
Comment (ThomsonTan) [inline sdk/src/version/version.cc:22]: `buildscripts/pre-commit` should update the `BUILD_DATE` here?

Comment (esigo) [reply sdk/src/version/version.cc]: thanks, updated

File: CHANGELOG.md (hunk 4 above)
Comment (esigo) [reply CHANGELOG.md:54]: thanks added the note:

Notes:

Metrics API/SDK GA release includes PRs [#1686](https://github.com/open-telemetry/opentelemetry-cpp/pull/1686), [#1701](https://github.com/open-telemetry/opentelemetry-cpp/pull/1701), and [#1707](https://github.com/open-telemetry/opentelemetry-cpp/pull/1707) with breaking changes in the Metrics API and SDK.

File: CHANGELOG.md (hunk 3 above)
Comment (esigo) [reply CHANGELOG.md]: thanks, added the tag

**Review by lalitb — APPROVED** [review]
Thank you, and nicely done - even though it was the first release :)

**Review by ThomsonTan — APPROVED** [review]
LGTM once the format is fixed. Thanks @esigo

File: CHANGELOG.md [hunk 6]
```diff
@@ -15,13 +15,57 @@ Increment the:
...
+* [DOCS] Document libthrift 0.12.0 doesn't work with Jaeger exporter [#1714](https://github.com/open-telemetry/opentelemetry-cpp/pull/1714)
+* [Metrics SDK] Add Monotonic Property to Sum Aggregation, and
+unit tests for Up Down Counter [#1675](https://github.com/open-telemetry/opentelemetry-cpp/pull/1675)
+* [Metrics SDK] Move Metrics Exemplar processing behind feature flag [#1710](https://github.com/open-telemetry/opentelemetry-cpp/pull/1710)
+* [Metrics API/SDK] Change Meter API/SDK to return nostd::unique_ptr
+ for Sync Instruments [#1707](https://github.com/open-telemetry/opentelemetry-cpp/pull/1707)
+which includes breaking change in the Metrics api and sdk.
+* [BUILD] Add e2e test to asan & tsan CI [#1670](https://github.com/open-telemetry/opentelemetry-cpp/pull/1670)
+* [BUILD] Add otlp-grpc example bazel [#1708](https://github.com/open-telemetry/opentelemetry-cpp/pull/1708)
+* Fix debug log of OTLP HTTP exporter and ES log exporter [#1703](https://github.com/open-telemetry/opentelemetry-cpp/pull/1703)
```
Comment (marcalff) [inline CHANGELOG.md:30]: Add [TRACE SDK] for 1703

**Review by marcalff — APPROVED** [review]
LGTM, see comment for changelog.

**Pull request merged**